include/hashinator/hashinator.h

```diff
@@ -106,39 +116,61 @@ class Hashmap {
    inline void set_status(status code) noexcept { _mapInfo->err = code; }
 
 public:
-   Hashmap() {
+   Hashmap():_allocator(Allocator{}) {
       preallocate_device_handles();
-      _mapInfo = _metaAllocator.allocate(1);
+      _mapInfo = reinterpret_cast<MapInfo*>(_allocator.allocate(get_number_of_Ts_for_Map_Info()));
+      *_mapInfo = MapInfo(5);
+      buckets = split::SplitVector<hash_pair<KEY_TYPE, VAL_TYPE>,Allocator>(
+          1 << _mapInfo->sizePower, hash_pair<KEY_TYPE, VAL_TYPE>(EMPTYBUCKET, VAL_TYPE()),_allocator);
+#ifndef HASHINATOR_CPU_ONLY_MODE
+      SPLIT_CHECK_ERR(split_gpuMemcpy(device_map, this, sizeof(Hashmap), split_gpuMemcpyHostToDevice));
+#endif
+   };
+   
+   Hashmap(const Allocator& allocator):buckets(split::SplitVector<hash_pair<KEY_TYPE, VAL_TYPE>,Allocator>(
+          1 << 5, hash_pair<KEY_TYPE, VAL_TYPE>(EMPTYBUCKET, VAL_TYPE()),_allocator)),_allocator(allocator){
```

Comment:
So default allocation of sizePower 5? Perhaps add a comment to those lines which use this value.